src/cmds/plots.py

```diff
@@ -173,30 +28,139 @@ def show(root_path):
+@click.option(
+    "-t",
+    "--tmp_dir",
+    help="Temporary directory for plotting files",
+    type=click.Path(),
+    default=Path("."),
+    show_default=True
+)
+@click.option("-2", "--tmp2_dir", help="Second temporary directory for plotting files", type=click.Path(), default=None)
+@click.option(
+    "-d",
+    "--final_dir",
+    help="Final directory for plots (relative or absolute)",
+    type=click.Path(),
+    default=Path("."),
+    show_default=True
+)
+@click.option("-i", "--plotid", help="PlotID in hex for reproducing plots (debugging only)", type=str, default=None)
+@click.option("-m", "--memo", help="Memo in hex for reproducing plots (debugging only)", type=str, default=None)
+@click.option("-e", "--nobitfield", help="Disable bitfield", default=False, is_flag=True)
+@click.option(
+    "-x",
+    "--exclude_final_dir",
+    help="Skips adding [final dir] to harvester for farming",
+    default=False,
+    is_flag=True
+)
+@click.pass_context
+def create_cmd(ctx: click.Context, size: int, num: int, buffer: int, num_threads: int, buckets: int, stripe_size: int,
+               alt_fingerprint: int, pool_contract_address: str, farmer_public_key: str, pool_public_key: str,
+               tmp_dir: str, tmp2_dir: str, final_dir: str, plotid: str, memo: str, nobitfield: bool,
+               exclude_final_dir: bool):
+    class Params(object):
+        size = size
+        num = num
+        buffer = buffer
+        num_threads = num_threads
+        buckets = buckets
+        stripe_size = stripe_size
+        alt_fingerprint = alt_fingerprint
+        pool_contract_address = pool_contract_address
+        farmer_public_key = farmer_public_key
+        pool_public_key = pool_public_key
+        tmp_dir = Path(tmp_dir)
+        tmp2_dir = Path(tmp2_dir)
+        final_dir = Path(final_dir)
+        plotid = plotid
+        memo = memo
+        nobitfield = nobitfield
+        exclude_final_dir = exclude_final_dir
+    create_plots(Params(), ctx.obj["root_path"])
+
+
+@plots_cmd.command('check', short_help="checks plots")
+@click.option("-n", "--num", help="Number of plots or challenges", type=int, default=None)
+@click.option(
+    "-g",
+    "--grep_string",
+    help="Shows only plots that contain the string in the filename or directory name",
+    type=str,
+    default=None
+)
+@click.option("-l", "--list_duplicates", help="List plots with duplicate IDs", default=False, is_flag=True)
+@click.option("--debug-show-memo", help="Shows memo to recreate the same exact plot", default=False, is_flag=True)
+@click.option("--challenge-start", help="Begins at a different [start] for -n [challenges]", type=int, default=None)
+@click.pass_context
+def check_cmd(ctx: click.Context, num: int, grep_string: str, list_duplicates: bool, debug_show_memo: bool,
+              challenge_start: int):
+    check_plots(ctx.obj["root_path"], num, challenge_start, grep_string, list_duplicates, debug_show_memo)
```

Comment:
This function is only explicitly called from here, so, maybe we can move it back to this file, like the rest of the CLI functions. (I don't know if there are other python code importing and using that function from the outside).

Reply:
I'd like to keep this modular for now, check plots seems useful for other purposes as well